core/modules/qdisp/Executive.cc

```diff
@@ -181,7 +160,7 @@ void Executive::add(int refNum, Executive::Spec const& s) {
 bool Executive::join() {
     // To join, we make sure that all of the chunks added so far are complete.
     // Check to see if _requesters is empty, if not, then sleep on a condition.
-    _waitUntilEmpty();
+    _reapUntilEmpty();
```

Comment:
prefer blockUntilDone instead of reapUntilEmpty. Reason:
waitUntilEmpty made sense to me, because the waiting was the most
important part. The reaping was a means of garbage collecting
previously-in-flight tasks. I understand that "wait" may not sound
appropriate because there is a meaning conflict, so how about "block",
because that's really what's going on, and "until done" because being
"done" is what's important, not that anything is empty or reaped. 

Your suggested "waitAll" is totally fine too.